src/Utils/Arrays.php

```diff
@@ -225,6 +225,7 @@ public static function flatten(array $array, bool $preserveKeys = false): array
 
 	/**
 	 * Checks if the array is indexed in ascending order of numeric keys from zero, a.k.a list.
+	 * @return ($value is list ? true : false)
```

Comment:
What's it good for?

Reply:
If variable is of type list it will always return true. Informs user that check is effectively useless.

But it should probably be `@return ($value is list ? true : bool)`

Reply:
I understand what it does, I ask what good it does :-) I just need an example.

Reply:
Here it is. `If condition is always true.`
https://phpstan.org/r/37d792a0-1fe7-488a-8818-15048810f6c0

Reply:
Sadly it is only half of the solution and it seems phpstan does not support assert for list. Even after isList() is given value array
https://phpstan.org/r/c26d6864-a46a-4d81-9fad-54dbc75d7e83

Reply:
Here is a better example. PHPStan does support asserts but the annotation needs to be prefixed.

https://phpstan.org/r/a7d26dd0-17a2-4cd7-abda-24e5d302d61d

Reply:
@enumag Thank you! I didn't know unprefixed version is not recognized...

Reply:
@enumag `@return` is probably pointless then, isn't it?

Reply:
@dg Yeah, it is indeed. I even checked Psalm. The return annotation doesn't make any difference for it either.

Reply:
@mabar @enumag Why do you think the return type should be `@return ($value is list ? true : **bool**)`? Can you give me example of value type which is not list, but the method will return true?

This change was actually the reason, why the examples you've posted seemingly did not work. The correct conditional return annotation is enough for phpstan to correctly detect always true/false conditions & perform type narrowing.

https://phpstan.org/r/c3f4cdc0-3417-42d8-9e5f-24f1c722b01c

Reply:
@xificurk Because $value can be a list without PHPStan being aware of it.

Reply:
@enumag Can you create example in phpstan playground?

btw, isn't it exactly the case of the last if-else block in [my example](https://phpstan.org/r/c3f4cdc0-3417-42d8-9e5f-24f1c722b01c)? PHPStan can't know if the value is list or not, but if it is, the `isList()` method will always return true, otherwise false.

Reply:
I tried a few things and yeah you're right, false is enough there. PHPStan does treat unknown array as a possible list.